### ISSUE TITLE
RE: turbojpeg-static.lib

### ISSUE THREAD

**Issue by tjenniges** [jwezorek/vs_pdfium#4]
Hi Joe,

Why is "turbojpeg-static.lib" statically linked to instead of building from sources like the other "third_party" libraries in source?

I like your github repo because it builds from sources that one can check (except for turbojpeg-static.lib").  I think google has inadvertently created security issues with pdfium by not putting out official builds of pdfium -- people not familiar with C/C++ download pre-built DLLs from Github and Nuget without knowing if they can trust the "person" building the DLLs.

Thank you.

**Comment by jwezorek** [comment]
Because,  as I understand it, turbojpeg can’t be built with out-of-the-box Visual Studio because some of its source code is assembly language but it uses the NASM dialect of assembly rather than the MASM dialect. Now you can install an extension to Visual Studio that will compile NASM, but if I required an extension then someone could not just pull the GitHub repo and press build and have it work.

Someone correct me if I am wrong on this...

The other thing we could do is just replace turbojpeg with regular libjpeg code and it would compile but would just be slower decoding Jpegs which imo would not be that big of a deal

**Comment by tjenniges** [comment]
I don't think it would be that big of a deal. What would be the cost, fractions of a millisecond per decoded image?

**Comment by jwezorek** [comment]
Right that’s my thinking too ... so maybe I’ll do that.

**Comment by 1000374** [comment]
When I compile it into a dynamic link library, it prompts me

1> LINK: fatal error LNK1104: Unable to open file 'turbojpeg static. lib'
